### PR TITLE
Convert grader-host to TypeScript

### PR DESCRIPTION
**Pull request by nwalters512**
Depends on #10049.

### CODE REVIEW

**Review by mylesw27 — APPROVED** [review]
The TS conversions look correct. I am unable to test locally but do not want to block on that.

**Comment by nwalters512** [comment]
I'll test this in our staging environment before merging.

File: apps/grader-host/src/index.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// @ts-check
+import { Readable } from 'node:stream';
```
Comment (jonatanschroeder) [inline apps/grader-host/src/index.ts:1]:
```suggestion
import type { Readable } from 'node:stream';
```

File: apps/grader-host/src/index.ts [hunk 2]
```diff
@@ -214,19 +211,14 @@ async function handleJob(job) {
...
-
-/**
- * @param {Context} context
- * @param {Date} receivedTime
- */
-async function reportReceived(context, receivedTime) {
+interface Context {
+  docker: Docker;
+  s3: S3;
+  logger: import('./lib/jobLogger.js').WinstonBufferedLogger;
```
Comment (jonatanschroeder) [inline apps/grader-host/src/index.ts:217]: Do you want to import this?

File: apps/grader-host/src/lib/receiveFromQueue.ts [hunk 3]
```diff
@@ -1,12 +1,11 @@
-// @ts-check
 import { setTimeout as sleep } from 'node:timers/promises';
 
 import {
   ReceiveMessageCommand,
   ChangeMessageVisibilityCommand,
   DeleteMessageCommand,
+  SQSClient,
```
Comment (jonatanschroeder) [inline apps/grader-host/src/lib/receiveFromQueue.ts:7]:
```suggestion
  type SQSClient,
```

File: apps/grader-host/src/lib/receiveFromQueue.ts [hunk 4]
```diff
@@ -84,41 +75,36 @@ async function startHeartbeat(sqs, queueUrl, receiptHandle) {
...
+  while (true) {
+    const data = await sqs.send(
+      new ReceiveMessageCommand({
+        MaxNumberOfMessages: 1,
+        QueueUrl: queueUrl,
+        WaitTimeSeconds: 20,
+      }),
+    );
+    const message = data.Messages?.[0];
+    if (!message || !message.Body) continue;
```
Comment (jonatanschroeder) [inline apps/grader-host/src/lib/receiveFromQueue.ts:89]: Did you mean `break`?

**Review by jonatanschroeder — APPROVED** [review]

File: apps/grader-host/src/index.ts (hunk 2 above)
Comment (nwalters512) [reply apps/grader-host/src/index.ts:217]: Yep - this was a bad merge conflict resolution.

File: apps/grader-host/src/lib/receiveFromQueue.ts (hunk 3 above)
Comment (nwalters512) [reply apps/grader-host/src/lib/receiveFromQueue.ts]: Done in d7e7403.

File: apps/grader-host/src/index.ts (hunk 2 above)
Comment (nwalters512) [reply apps/grader-host/src/index.ts]: Fixed in d7e7403.

File: apps/grader-host/src/index.ts (hunk 1 above)
Comment (nwalters512) [reply apps/grader-host/src/index.ts]: Done in d7e7403.

**Pull request merged**